scripts/secureboot-certs

```diff
@@ -1,160 +1,522 @@
+    It is API-equivalent with the Host class.
+    """
+
+
+def print_cert(path, uuid, klass):
+    print("\tCertificate: %s" % os.path.basename(path))
+    print("\t%s: %s" % (klass, uuid))
+    print("\tAuth file md5: %s" % hashfile(path))
+    print("\tData:")
+    output = subprocess.check_output(["hexdump", "-Cv", path])
+    for line in output.splitlines():
+        print("\t\t%s" % line)
+    print("")
+
+
+def report(session):
+    print("\n{} -- Report".format(os.path.basename(sys.argv[0])))
+    pool = Pool.get_all(session)[0]
+    paths = pool.save_certs_to_disk()
+    print("Certificate Info for pool: %s):" % pool.uuid)
+    s = "\tCertificates (%s): " % len(paths)
+    s += ", ".join(os.path.basename(p) for p in paths)
+    s += "\n"
+    print(s)
+    for path in paths:
+        print_cert(path, pool.uuid, klass="Pool")
+
+    hosts = Host.get_all(session)
+    print("Hosts(%s): %s" % (len(hosts), ", ".join(h.uuid for h in hosts)))
+    for host in hosts:
+        paths = host.save_certs_to_disk()
+        print("Certificate Info for host: %s):" % host.uuid)
+        s = "\tCertificates (%s): " % len(paths)
+        s += ", ".join(os.path.basename(p) for p in paths)
+        s += "\n"
+        print(s)
+        for path in paths:
+            print_cert(path, host.uuid, klass="Host")
+        print("")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description=("Configure guest UEFI certificates for an XCP-ng system.")
+    )
+
+    parser.add_argument(
+        "--verbose",
+        action="store_true",
+        help="Verbose output.",
+    )
+
+    action_parsers = parser.add_subparsers()
+    install_parser = action_parsers.add_parser(
+        Actions.INSTALL,
+        help="Install UEFI certificates to every host in the pool.",
+        description="Install UEFI certificates to every host in the pool.",
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        epilog="""Important Note: all Microsoft certs are downloaded
+automatically from Microsoft's server, and therefore require network access.
+
+Certificate / auth file URLs:
+CA: {}
+PCA: {}
+KEK: {}
+dbx: {}
+""".format(
+            Urls.CA, Urls.PCA, Urls.KEK, Urls.dbx
+        ),
+    )
+    install_parser.set_defaults(action=Actions.INSTALL)
+
+    install_parser.add_argument(
+        "PK",
+        metavar="PK",
+        help="'default' for the default XCP-ng PK or a path to a custom auth file.",
+    )
+    install_parser.add_argument(
+        "KEK",
+        metavar="KEK",
+        help="'default' for the default Microsoft certs or a path to a custom auth file.",
+    )
+    install_parser.add_argument(
+        "db",
+        metavar="db",
+        help="'default' for the default Microsoft certs or a path to a custom auth file.",
+    )
+    install_parser.add_argument(
+        "dbx",
+        metavar="dbx",
+        help="'none' for no dbx, 'default' for the most recent UEFI dbx, or a path to a custom auth file.",
```

Comment:
can we replace `default` with `latest` here since a default ought to be safe and in this case it isn't?

Reply:
And maybe explain in the help string why latest is both better for security AND more riskier for some existing systems?

Reply:
And also explain that even if you don't have the latest dbx, each VM is able to update its own dbx (like Windows does through its security updates) that overrides the one configured here?

Reply:
Changed this up.  Let me know what you think.

Reply:
I validated the newest commits with dbx set to `latest` and all looks good.  Ready when you are for merge.